plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcMetadataSessionProperties.java

```diff
@@ -46,6 +47,11 @@ public JdbcMetadataSessionProperties(JdbcMetadataConfig jdbcMetadataConfig, @Max
                         "Enable aggregation pushdown",
                         jdbcMetadataConfig.isAggregationPushdownEnabled(),
                         false))
+                .add(booleanProperty(
```

Comment:
Would it be possible to have the config and session toggle present only for these connectors where it matters?

This can be as simple as a separate guice module providing both of them.
Then `PredicatePushdownController#CASE_INSENSITIVE_CHARACTER_PUSHDOWN` will inspect the session toggle, and thus every connector using `CASE_INSENSITIVE_CHARACTER_PUSHDOWN` will be validated to have the module installed.